builtin/method_list.py

```diff
@@ -92,7 +92,7 @@ def Call(self, rd):
         rd.Done()
         i = 0
         while i < len(li):
-            if needle.tag() == li[i].tag() and val_ops.ExactlyEqual(li[i], needle, loc.Missing):
+            if val_ops.ExactlyEqual(li[i], needle, loc.Missing):
```

Comment:
This looks good, can you try this:

```
$ bin/ysh -c '= [1.0, 2.0] => indexOf(3.14)'
  = [1.0, 2.0] => indexOf(3.14)
  ^
[ -c flag ]:1: fatal: Equality isn't defined on Float
```

And then try changing loc.Missing to `rd.LeftParenToken()`

I think it will improve the error

---

And perhaps add this test case to `test/ysh-runtime-errors.sh`

It could be in a new function at the end or some existing function

That is how we "eyeball" all the errors to make sure they are decent

Reply:
I was wondering which var we (currently?) can't compare for equality.

And thanks, I didn't see any other code getting a location from `rd` so that's why I went with `Missing`. Definitely better with a proper location!

Reply:
I added the error to the existing `test-fat-arrow` func but maybe it's better to have an additional func `test-ysh-methods`? But I guess we can change this if more methods require a testcase?